### DISCUSSION TITLE
Missing symbols when using extra C++ sources

### DISCUSSION THREAD

**Discussion by jneem** [tree-sitter/tree-sitter#1261] [Q&A] [answered]
The tl;dr is that if you check out [this](https://github.com/jneem/tree-sitter-latex) latex parser and run `tree-sitter test` then it fails with

```
Error opening dynamic library "/home/user/.cache/tree-sitter/lib/latex.so"

Caused by:
    /home/user/.cache/tree-sitter/lib/latex.so: undefined symbol: _ZN5LaTeX7Scanner8keywordsB5cxx11E
```

This seems to be related to the fact that there are multiple C++ source files in `src/`, not just `scanner.cc`; for example, the missing symbol above is defined in `scanner_keywords.cc`. Note that these extra source files are listed in `binding.gyp`, and if I run `node-gyp build` then it does compile these files. However, it seems that running `tree-sitter test` does not compile these files: it only compiles `scanner.cc` and the symbols from the other files are missing.

If I concatenate all my source files together (by uncommenting the lines at the end of `scanner.cc`) then `tree-sitter test` works (but running `node-gyp build` now fails because of duplicate symbols).

Is there a supported way to have tree-sitter build multiple C++ source files, instead of just `scanner.cc`?

**Comment by ahelwer** [comment, marked as answer]
Not really an answer but this seems related to this issue: https://github.com/tree-sitter/tree-sitter/issues/1246

**Reply by jneem** [reply]
Thanks! Based on that, it seems that the answer is no: `scanner.cc` is hard-coded, and there's no support for multiple source files. I'll open an issue and see if they're open to a PR about that.

**Reply by ahelwer** [reply]
@jneem are you looking at taking over development of that latex grammar? There's [another one](https://github.com/latex-lsp/tree-sitter-latex) that is currently the canonical latex grammar for neovim's tree-sitter plugin but when I tried it it definitely [left me wanting](https://github.com/latex-lsp/tree-sitter-latex/issues/6).

**Reply by jneem** [reply]
"taking over development" would be overstating it, but at least I'm evaluating it to see how much effort it would take to get it working reasonably. I also saw the latex-lsp one, and I also felt that it was incomplete.
